overlord/snapstate/backend/setup.go

```diff
@@ -100,7 +106,11 @@ func (b Backend) RemoveSnapFiles(s snap.PlaceInfo, typ snap.Type, meter progress
 	if _, err := os.Lstat(snapPath); err == nil {
 		// remove the kernel assets (if any)
 		if typ == snap.TypeKernel {
```

Comment:
I would have expected this check to go away, be folded in the lookup

Reply:
tbh I didn't even look. I will.